enhancements/sig-policy/74-order-policy-execution/README.md

```diff
@@ -0,0 +1,250 @@
+must be met in order to evaluate a policy template, which would allow the governance
+framework to automatically handle cases where policy templates need to be processed in
+order.
+
+## Motivation
+
+Currently, users have to split up policy templates that depend on others into multiple
+policies and manually create them in order. This is more resource intensive as users
+can have to create duplicate policies and more time consuming if users have to manually
+manage policy processing order. If a user elects to group the templates in one policy,
+there is no guarantee the templates will be processed in order, which could leave
+the policy in a bad state if a required resource has not yet been processed. For example,
+when using a policy to install an operator, if some objects are created before others it
+can cause the operator to not be installed properly. Allowing users to specify policy
+dependencies will alleviate these issues.
+
+### Goals
+
+1. Policy template dependencies: allow users to automatically process policy templates
+in a desired order and guarantee that this order will be stable
+2. Opt-in: any extra fields that are used to specify policy ordering should be optional,
+and policies that do not contain these fields should function as they did before in order
+to avoid issues with older policies
+
+### Non-Goals
+
+1. Status message matching: we plan to only allow users to conditionally process policy
+templates based on compliance state, so users will not be able to specify that they want a
+policy to be processed if another policy has a specific status message
+
+## Proposal
+
+We would add an optional new `dependencies` field to the spec on the configuration 
+policy level as well as an `extraDependencies` field on the template level to
+allow users to specify conditions for policy processing on both the policy and
+template level. When these fields are set and their conditions are not met, 
+the policy templates will not be applied on the managed cluster and the affected
+policy will enter a "pending" state. If this happens, any policy sets that contain
+the pending policy will also be marked as pending. When the conditions are met or
+the fields are not set, the policy templates will be processed normally.
+
+### User Stories
+
+1. As an author of a ConfigurationPolicy, I don’t want resources to be wasted when
+I know one object depends on another. If an object cannot be cleanly applied before
+another is created, I want to specify this in my Policy, and not waste
+resources (CPU, k8s API requests) in the policy controllers.
+2. As an author of a ConfigurationPolicy, I want to specify the order resources are
+applied in a policy when the API is not declarative. For some APIs (eg OLM), if
+resources are applied in the wrong order they can end up in an unhealthy state
+that I want to avoid.
+3. As a user of PolicySets, I want policies inside a set to be applied in a
```

Comment:
If one policy is pending state, what's status of the policyset ?

Reply:
I think it should follow the same rules as the Policies, unless there's a reason to handle it differently? So if any policies in the policyset are NonCompliant, the status will be NonCompliant (regardless of whether any are pending). If the policyset only has Compliant and Pending policies, then it will be Pending. And only if all the policies are Compliant will it be Compliant.

Reply:
OK. I think the priority of states is reasonable.   Can you add the warning message or tip message to prompt in the policyset card if any policy is pending in this policyset?